styles/Canonical/013-Spell-out-numbers-below-10.yml

```diff
@@ -9,13 +9,13 @@ scope:
   - table.cell
 level: warning
 swap:
-  (?<![\-\.\,:\/*-+])0(?![\-\.\,:\/*-+]): zero
-  (?<![\-\.\,:\/*-+])1(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): one
-  (?<![\-\.\,:\/*-+])2(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): two
-  (?<![\-\.\,:\/*-+])3(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): three
-  (?<![\-\.\,:\/*-+])4(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): four
-  (?<![\-\.\,:\/*-+])5(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): five
-  (?<![\-\.\,:\/*-+])6(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): six
-  (?<![\-\.\,:\/*-+])7(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): seven
-  (?<![\-\.\,:\/*-+])8(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): eight
-  (?<![\-\.\,:\/*-+])9(?![\-\.\,:\/*-+])(?! Jan(?!uary)?| Feb(?!ruary)?| Mar(?!ch)?| Apr(?!il)?| May| Jun(?:e)?| Jul(?:y)?| Aug(?:ust)?| Sep(?:tember)?| Oct(?:ober)?| Nov(?:ember)?| Dec(?:ember)?): nine
+  (?<![\-\.\,:\/*-+#]|Day\s)0(?![\-\.\,:\/*-+]): zero
```

Comment:
Apologies for the drive-by. :)

```
*-+
```

I presume we want each of these to be literals, and therefore `*\-+`. Currently this provides a range from `*` (U+002A) to `+` (U+002B).

Reply:
Quite right. It should be:
`\-\.\,:\/\*\-\+#`

It needs to apply to all of them, for both lookups.

Reply:
hey @SecondSkoll ,

Apologies, but I'm confused with your comment.
`\-\.\,:\/\*\-\+#`

Isn't the first `\-` already covering the hyphen character? And that is repeated in this `\*\-\+` part?

Thanks.

Reply:
Actually, yes. I wasn't really looking at the first part. Silly of me. I suppose it would make no difference then, given that `*-+` would all be covered.

Reply:
Given what has been said above I can assume no changes are needed here. 
@medubelko Is that so?
P.S. If this is still up for debate please consider opening a new issue as it is slightly out of scope for this PR.

Reply:
It's extra noise in the expression, but since this was present in the original, feel free to ignore me.

Reply:
The current implementation is poor (mea culpa), but functional. No changes are **necessary**, but if we want to optimise it, the options are either to:

- display each character captured by delimiting the necessary ones (so everyone looking at the rule knows what it's looking for), or
- capture a large range of characters to minimise the expression.

So, either:

- `[\*\+\,\-\.\/:#]` which lists all relevant characters, or
- `[*-\/:#]` which is U+002A through U+002F (`*+,-./`), `:` and `#`.

Potentially, optimising our rules could be left to another epic, as I'm sure we have suboptimal implementations in the other rules.

Reply:
Let's not expand the scope of this PR with refactoring. I've created an issue covering the regex optimization - https://github.com/canonical/praecepta/issues/73. Please use it to comment or add your suggestions on the matter.